aspnetcore/log-mon/metrics/metrics.md

```diff
@@ -175,6 +175,55 @@ Alternatively, enter counter category such as `kestrel` in the **Expression** in
 
 ![dashboard-screenshot2](~/log-mon/metrics/metrics/static/dashboard-screenshot.png)
 
+## Test metrics in ASP.NET Core apps
+
+It's possible to test metrics in ASP.NET Core apps.
+
+ASP.NET Core uses the `IMeterFactory` API to create and obtain meters. The meter factory integrates metrics with dependency injection, making isolating and collecting metrics easy. `IMeterFactory` is especially useful for unit testing, where multiple tests can run side-by-side and you want to only gather data for your test.
+
+```cs
+public class BasicTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly WebApplicationFactory<Program> _factory;
+    public BasicTests(WebApplicationFactory<Program> factory) => _factory = factory;
+
+    [Fact]
+    public async Task Get_RequestCounterIncreased()
+    {
+        // Arrange
+        var client = _factory.CreateClient();
+        var meterFactory = _factory.Services.GetRequiredService<IMeterFactory>();
+        var collector = new MetricCollector<double>(meterFactory,
+            "Microsoft.AspNetCore.Hosting", "http.server.request.duration");
+
+        // Act
+        var response = await client.GetAsync("/");
+
+        // Assert
+        Assert.Equal("Hello World!", await response.Content.ReadAsStringAsync());
+
+        await collector.WaitForMeasurementsAsync(minCount: 1);
+        Assert.Collection(collector.GetMeasurementSnapshot(),
+            measurement =>
+            {
+                Assert.Equal("http", measurement.Tags["url.scheme"]);
+                Assert.Equal("GET", measurement.Tags["http.request.method"]);
+                Assert.Equal("/", measurement.Tags["http.route"]);
+            });
+    }
+}
+```
+
+The proceeding test:
+
+* Bootstraps a web app in memory with <xref:Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory%601>. `Program` in the factory's generic argument specifies the web app.
+* Collects metrics values with <xref:Microsoft.Extensions.Telemetry.Testing.Metering.MetricCollector<T>>.
```

Comment:
That's a valid link but is breaking the build for some reason. @tdykstra ?
you could use
[`MetricCollector<T>`](/dotnet/api/microsoft.extensions.telemetry.testing.metering.metriccollector-1.-ctor?view=dotnet-plat-ext-8.0#microsoft-extensions-telemetry-testing-metering-metriccollector-1-ctor(system-diagnostics-metrics-instrument((-0))-system-timeprovider))

Reply:
It wasn't a valid link -- it needed `%601` instead of `<T>`